### PR TITLE
Throw error when no characters specified

### PR DESCRIPTION
**Pull request by Lex-Ashu**
Added Error Handling in Password Generator.

### CODE REVIEW

File: passwd/__init__.py [hunk 1]
```diff
@@ -182,7 +182,10 @@ def generate(
             allowed_chars += punctuation if special else ""
         elif self.special:
             allowed_chars += punctuation
-
+        if not allowed_chars:
+            raise ValueError("No character types selected for password generation.")     #Error Handling
+        
+        return "".join(choice(allowed_chars) for _ in range(length))
```
Comment (bsoyka) [inline passwd/__init__.py:188]: This line is duplicated now—let's remove one of them.

File: passwd/__init__.py [hunk 2]
```diff
@@ -182,7 +182,10 @@ def generate(
             allowed_chars += punctuation if special else ""
         elif self.special:
             allowed_chars += punctuation
-
+        if not allowed_chars:
+            raise ValueError("No character types selected for password generation.")     #Error Handling
```
Comment (bsoyka) [inline passwd/__init__.py:186]: Can you add a test case for this in `tests/test_password_generator.py`?

**Review by bsoyka — CHANGES_REQUESTED** [review]
Thanks for contributing! There are a couple of minor things to work on here—see my comments for more.

File: passwd/__init__.py (hunk 2 above)
Comment (Lex-Ashu) [reply passwd/__init__.py:186]: I will resolve it till that can u mark it for hacktober accepted as it is urgent and I will surely update the code with your issues

**Comment by Lex-Ashu** [comment]
I have removed the duplicated line and I have added the required test case
in test_password_generator.py. Have a look at it and let me know if any
changes are needed or I would be grateful if you merge both of my pull
requests.
Thank You

**Comment by bsoyka** [comment]
Both the main change and the tests need to be in the same pull request, so the test runs on the updated code and we can make sure everything's working properly.

**Comment by Lex-Ashu** [comment]
Done

File: tests/test_password_generator.py [hunk 3]
```diff
@@ -18,6 +18,9 @@ def test_length():
     assert len(all_0) == 0
     assert len(all_0.generate()) == 0
 
+def test_no_character_types_selected():    #Added a test case when all character types are set to False
+    with raises(ValueError,match="No character types selected for password genration."):
```
Comment (bsoyka) [inline tests/test_password_generator.py:22]: You have a typo here (`genration`) which is causing the test to fail.

**Review by bsoyka — CHANGES_REQUESTED** [review]

**Review by bsoyka — APPROVED** [review]
Tests are passing now—looks good to me! Thanks for your contributions!

**Pull request merged**